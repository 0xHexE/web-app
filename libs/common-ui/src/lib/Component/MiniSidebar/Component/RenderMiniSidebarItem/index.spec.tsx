import { getAllByTestId, render } from '@testing-library/react';
import { RenderMiniSidebarItem } from './index';
import { Menu16 } from '@carbon/icons-react';

describe('RenderMiniSidebarItem', function () {
  it('should render mini sidebar item', function () {
    const { baseElement: baseElement2 } = render(
      <RenderMiniSidebarItem
        item={{ kind: 'action', icon: <Menu16 />, key: 'test', title: 'test' }}
        isLastItem={false}
      />
    );
    expect(baseElement2).toMatchSnapshot();

    const { baseElement } = render(
      <RenderMiniSidebarItem
        item={{ kind: 'action', icon: <Menu16 />, key: 'test', title: 'test' }}
        isLastItem={true}
      />
    );

    expect(baseElement).toMatchSnapshot();

    const { baseElement: baseElement3 } = render(
      <RenderMiniSidebarItem
        item={{ kind: 'spacer', key: 'test' }}
        isLastItem={true}
      />
    );

    expect(baseElement3).toMatchSnapshot();

    const { container } = render(
      <RenderMiniSidebarItem
        item={{ kind: 'action', key: 'test2', icon: <Menu16 />, title: 'test' }}
        isLastItem={true}
      />
    );

    expect(baseElement3).toMatchSnapshot();

    getAllByTestId(container, 'render-mini-sidebar-item');
  });
});
