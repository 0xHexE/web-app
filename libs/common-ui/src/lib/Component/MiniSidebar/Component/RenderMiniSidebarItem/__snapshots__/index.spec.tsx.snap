// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderMiniSidebarItem should render mini sidebar item 1`] = `
<body>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        data-testid="render-mini-sidebar-item"
        id="1__anchor"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;

exports[`RenderMiniSidebarItem should render mini sidebar item 2`] = `
<body>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        data-testid="render-mini-sidebar-item"
        id="1__anchor"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        data-testid="render-mini-sidebar-item"
        id="2__anchor"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;

exports[`RenderMiniSidebarItem should render mini sidebar item 3`] = `
<body>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        data-testid="render-mini-sidebar-item"
        id="1__anchor"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        data-testid="render-mini-sidebar-item"
        id="2__anchor"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div>
    <span
      class=""
    />
  </div>
</body>
`;

exports[`RenderMiniSidebarItem should render mini sidebar item 4`] = `
<body>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        data-testid="render-mini-sidebar-item"
        id="1__anchor"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        data-testid="render-mini-sidebar-item"
        id="2__anchor"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div>
    <span
      class=""
    />
  </div>
  <div>
    <div
      class=""
      data-baseweb="block"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class=""
        data-baseweb="button"
        data-testid="render-mini-sidebar-item"
        id="3__anchor"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 12H14V13H2zM2 9H14V10H2zM2 6H14V7H2zM2 3H14V4H2z"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;
